pkg/pkgupdates/pkgupdates.jsx

```diff
@@ -0,0 +1,558 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var cockpit = require("cockpit");
+var React = require("react");
+var moment = require("moment");
+var Tooltip = require("cockpit-components-tooltip.jsx").Tooltip;
+require("listing.less");
+
+const _ = cockpit.gettext;
+
+// "available" heading is built dynamically
+const STATE_HEADINGS = {
+    "loading": _("Loading available updates, please wait..."),
+    "locked": _("Some other program is currently using the package manager, please wait..."),
+    "uptodate": _("No updates available, system is up to date"),
+    "applying": _("Applying updates"),
+    "updateError": _("Applying updates failed"),
+    "loadError": _("Loading available updates failed"),
+}
+
+// see https://github.com/hughsie/PackageKit/blob/master/lib/packagekit-glib2/pk-enum.h
+const PK_EXIT_ENUM_SUCCESS = 1;
+const PK_EXIT_ENUM_FAILED = 2;
+const PK_ROLE_ENUM_REFRESH_CACHE = 13;
+const PK_ROLE_ENUM_UPDATE_PACKAGES = 22;
+const PK_INFO_ENUM_SECURITY = 8;
+const PK_STATUS_ENUM_WAIT = 1;
+const PK_STATUS_ENUM_UPDATE = 10;
+const PK_STATUS_ENUM_WAITING_FOR_LOCK = 1;
+
+const PK_STATUS_STRINGS = {
+    9: _("Downloading"),
```

Comment:
Yes, me neither, the download part is reported as `INSTALL` on Fedora. But I think it can't hurt to provide a human readable string for it anyway - it may get fixed in PackageKit one day, or already be used with other backends (yum, apt, etc.)